Sources/ContextPropagation/Instrumentation/MultiplexInstrumentationMiddleware.swift

```diff
@@ -0,0 +1,21 @@
+public struct MultiplexInstrumentationMiddleware<InjectInto, ExtractFrom> {
+    private var middleware: [InstrumentationMiddleware<InjectInto, ExtractFrom>]
+
+//    public init(_ middleware: [InstrumentationMiddleware<InjectInto, ExtractFrom>]) {
+//        self.middleware = middleware
+//    }
+
+    public init<M: InstrumentationMiddlewareProtocol>(_ middleware: [M]) where M.InjectInto == InjectInto, M.ExtractFrom == ExtractFrom {
```

Comment:
I think I'd suggest doing the following style everywhere for such long where clauses:

```swift
    public init<Middleware>(_ middleware: Middleware)
        where
        Middleware: InstrumentationMiddlewareProtocol,
        Middleware.ExtractFrom == Self.ExtractFrom,
        Middleware.InjectInto == Self.InjectInto {
```

it reads much better 😄